Sources/JWTKeychain/Support/Configuration/Configuration.swift

```diff
@@ -15,67 +15,103 @@ public protocol ConfigurationType {
+    }
+
+    public init(drop: Droplet) throws {
+        try self.init(config: drop.config)
+    }
+
+    public init(config: Config) throws {
+
+        guard let jwtConfig = config["jwt"]?.object else {
+            throw Error.noJWTConfig
+        }
+
+        guard let signer: String = jwtConfig["signer"]?.string else {
+            throw Error.missingConfig("signer")
+        }
+
+        guard let secondsToExpire = jwtConfig["secondsToExpire"]?.double else {
+            throw Error.missingConfig("secondsToExpire")
+        }
+
+        guard let signatureKey = jwtConfig["signatureKey"]?.string else {
+            throw Error.missingConfig("signatureKey")
+        }
+
+        let publicKey: String? = jwtConfig["publicKey"]?.string
+
+        if publicKey == nil {
+            //The ECDSA and RSA (ES*/RS*) signers take a private key for signing and needs a matching public key for verifying.
+            if signer.hasPrefix("ES") || signer.hasPrefix("RS"){
+                throw Error.missingConfig("publicKey")
+            }
+        }
+
+        self.init(signer: signer, signatureKey: signatureKey, publicKey: publicKey, secondsToExpire: secondsToExpire)
+
+    }
+
+    public init(signer: String, signatureKey: String, publicKey: String?, secondsToExpire: Double){
         self.signer = signer
         self.signatureKey = signatureKey
         self.publicKey = publicKey
-        self.secondsToExpire = secondsToExpire}
-    // Register configs
-    public static func boot(signer: String, signatureKey: String, publicKey: String?, secondsToExpire: Double) throws {
-        Configuration.instance = Configuration(signer: signer, signatureKey: signatureKey, publicKey: publicKey, secondsToExpire: secondsToExpire)
-    }
-    
-    
+        self.secondsToExpire = secondsToExpire
+      }
+
     /// Gets token signature key
     ///
     /// - Returns: signature key
-    public static func getTokenSignatureKey() -> Bytes {
-        
-        return Array(Configuration.instance!.signatureKey.utf8)
-        
+    public func getTokenSignatureKey() -> Bytes {
+        return Array(self.signatureKey.utf8)
+
     }
-    
+
     /// Gets token public key
     ///
     /// - Returns: public key
     /// - Throws: if cannot retrieve signature key
-    public static func getTokenPublicKey() throws -> Bytes {
-        return Array(Configuration.instance!.publicKey!.utf8)
-        
+    public func getTokenPublicKey() throws -> Bytes {
+        return Array(self.publicKey!.utf8)
+
     }
-    
+
     /// Generates the expiration date based on the
     /// configured seconds to expire
     ///
     /// - Returns: token expiration date
     /// - Throws: on unable to create the date
-    public static func generateExpirationDate() throws -> Date {
-        
-        return Date() + Configuration.instance!.secondsToExpire!
-        
+    public func generateExpirationDate() throws -> Date {
+        return Date() + self.secondsToExpire!
```

Comment:
`secondsToExpire` should no longer be optional and the force unwrap should be removed.